packages/devtools_app/lib/src/screens/inspector_v2/inspector_controller.dart

```diff
@@ -731,6 +731,25 @@ class InspectorController extends DisposableController
 
     try {
       final newSelection = await pendingSelectionFuture;
+
+      // Show an error and don't update the selected node in the tree if the
+      // user selected an implementation widget in the app while implementation
+      // widgets are hidden in the tree.
+      if (implementationWidgetsHidden.value && newSelection != null) {
+        final isInTree =
+            valueToInspectorTreeNode.containsKey(newSelection.valueRef);
+        final hasParent = newSelection.parent != null;
+        final isImplementationWidget = !isInTree && !hasParent;
+        if (isImplementationWidget) {
+          notificationService.pushError(
+            'Selected an implementation widget. Please toggle "Show Implementation Widgets" and re-select it to inspect.',
```

Comment:
wording suggestion to make it clear where the "reselect" should happen:

...Please toggle "Show Implementation Widgets" and try to select a widget from the device again.